### PR TITLE
Second S3 endpoint: us-east-1

### LINKED ISSUE #788: (R) add addl AWS regional endpoint
- new worker class with a separate queue ("queue_as")
    - extend the other AWS class and use a diff queue name

- add new delivery worker as a value in the endpoint deliver class enum (NOTE:  worker is getting creds supplied via ENV;  we would need to specify ENV var at worker startup (e.g. rake take invocation - `rake resque:work`)

- add an endpoint that has the enum value and is named appropriately

- get creds for AWS new region into puppet (can be split out into separate ticket)

### PR DESCRIPTION
**Pull request by atz**
closes #788

### CODE REVIEW

File: spec/jobs/s3_east_delivery_job_spec.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+describe S3EastDeliveryJob, type: :job do
```
Comment (SaravShah) [inline spec/jobs/s3_east_delivery_job_spec.rb]: if you run `be rspec spec/job/s3_east_delivery_job_spec.rb` by it self, it fails. is it bc this is inheriting from s3_west, so it needs to run s3_west_spec before the s3_east_spec is ran?

File: spec/models/endpoint_spec.rb [hunk 2]
```diff
@@ -1,48 +1,36 @@
...
+      :endpoint,
       endpoint_name: 'aws-us-east-2',
-      endpoint_type_id: endpoint_type.id,
+      endpoint_type: endpoint_type,
       endpoint_node: 's3.us-east-2.amazonaws.com',
       storage_location: 'sdr-bucket-01'
     )
   end
 
-  it 'is not valid without valid attributes' do
```
Comment (SaravShah) [inline spec/models/endpoint_spec.rb]: 👍

File: spec/models/endpoint_spec.rb [hunk 3]
```diff
@@ -175,15 +161,10 @@
...
-    it 'does not allow SQL injection' do
-      # a bit too whitebox, but oh well...  specifically testing version field
-      # because we know we're not using the usual AR params for bind vars approach
-      # in the subquery.  we cast version.to_i, which will silently convert anything
-      # it can't figure out how to properly translate into zero.
-      bogus_version = "0)))); GRANT ALL PRIVILEGES ON DATABASE pres TO icanhaznjctionattk; DROP TABLE students; --"
-      generated_sql = Endpoint.which_need_archive_copy(druid, bogus_version).to_sql
-      expect(generated_sql).not_to match(/#{Regexp.escape(bogus_version)}/)
-      expect(generated_sql).to match(/"preserved_copies"."version" = 0/)
+    it 'Casts version to integer' do
```
Comment (ndushay) [inline spec/models/endpoint_spec.rb]: I would like to at least retain a comment about why we care (avoid SQL injection b/c if the version is being cast to an integer, it isn't possible to deliver any valuable SQL payload via it) in the code itself.

File: spec/jobs/s3_east_delivery_job_spec.rb (hunk 1 above)
Comment (atz) [reply spec/jobs/s3_east_delivery_job_spec.rb]: ah, just forgot to require `rails_helper`, thx

**Comment by atz** [comment]
Fixed both of those points, thanks.

File: spec/models/endpoint_spec.rb [hunk 4]
```diff
@@ -175,15 +161,11 @@
...
-    it 'does not allow SQL injection' do
-      # a bit too whitebox, but oh well...  specifically testing version field
-      # because we know we're not using the usual AR params for bind vars approach
-      # in the subquery.  we cast version.to_i, which will silently convert anything
-      # it can't figure out how to properly translate into zero.
-      bogus_version = "0)))); GRANT ALL PRIVILEGES ON DATABASE pres TO icanhaznjctionattk; DROP TABLE students; --"
-      generated_sql = Endpoint.which_need_archive_copy(druid, bogus_version).to_sql
-      expect(generated_sql).not_to match(/#{Regexp.escape(bogus_version)}/)
-      expect(generated_sql).to match(/"preserved_copies"."version" = 0/)
+    # since we build AREL subquery, the cast is a guarantee against SQL injection
```
Comment (ndushay) [inline spec/models/endpoint_spec.rb]: 👍

**Pull request merged**